### PR TITLE
feat(radio): add 'block' argument

### LINKED ISSUE #50790: Radio.Group希望新增一个block的属性，类似Segmented的Block效果
### What problem does this feature solve?

在使用Radio.Button的分组时，经常会用到适合父级宽度的单选组，效果如下：

### What does the proposed API look like?

```jsx
<Radio.Group
  options={[
    {
      label: '全部',
      value: 'all',
    },
    {
      label: '未读',
      value: 'unread',
    },
    {
      label: '已读',
      value: 'read',
    },
  ]}
  optionType="button"
  buttonStyle="solid"
  block // <-
/>
```

### PR DESCRIPTION
**Pull request by yuanliu147**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [x] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

radio.Group 包裹 radio.Button 时，支持自动撑开：close https://github.com/ant-design/ant-design/issues/50790

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Add the block attribute to make the Radio.Group automatically expand when it wraps the Radio.Button      |
| 🇨🇳 Chinese |     添加 block 属性，让 Radio.Group 包裹 Radio.Button 时自动撑开      |

### CODE REVIEW

**Comment by yuanliu147** [comment]
> 文档 API `Block` 漏了～

是哦，，我改下

File: components/radio/group.tsx [hunk 1]
```diff
@@ -111,6 +112,7 @@ const RadioGroup = React.forwardRef<HTMLDivElement, RadioGroupProps>((props, ref
     {
       [`${groupPrefixCls}-${mergedSize}`]: mergedSize,
       [`${groupPrefixCls}-rtl`]: direction === 'rtl',
+      [`${groupPrefixCls}-flex`]: block,
```
Comment (afc163) [inline components/radio/group.tsx:115]:
```suggestion
      [`${groupPrefixCls}-block`]: block,
```

**Comment by yuanliu147** [comment]
> > radio.Group 包裹 radio.Button 时，支持自动撑开
> 
> 包裹普通的 radio 可以一起实现么 🤔？

没问题，确实遗漏了。。。

File: components/radio/group.tsx [hunk 2]
```diff
@@ -120,7 +122,7 @@ const RadioGroup = React.forwardRef<HTMLDivElement, RadioGroupProps>((props, ref
   );
 
   const memoizedValue = React.useMemo<RadioGroupContextProps>(
-    () => ({ onChange: onRadioChange, value, disabled, name, optionType }),
+    () => ({ onChange: onRadioChange, value, disabled, name, optionType, block }),
     [onRadioChange, value, disabled, name, optionType],
```
Comment (li-jia-nan) [inline components/radio/group.tsx:126]: 这里是不是漏了 block

Comment (yuanliu147) [reply components/radio/group.tsx]: 是的...  [哭]

**Comment by yuanliu147** [comment]
> 这个是新特性，提交到 feature 分支吧

好的，，我换个pr

**Comment by li-jia-nan** [comment]
master 的 commit 不应该带过去，rebase 一下应该就好了

**Comment by yuanliu147** [comment]
> commit

好的，，我试试

**Comment by yuanliu147** [comment]
> master 的 commit 不应该带过去，rebase 一下应该就好了

大佬，rebase的话貌似有冲突，我重新提交一个pr吧，，

**Pull request closed without merging**